### ISSUE TITLE
Incorrect Screen Reader focus on transition

### ISSUE THREAD

**Issue by niksawtschuk** [react-navigation/stack#291]
## Context
While auditing our application using VoiceOver on iOS, I noticed that screen transitions (pushing on a stack as well as presenting modally) don't yield behaviour I would expect from a native iOS application using UINavigation and UIViewControllers.

At minimum, I would expect that navigating to a new screen (either by pushing one on the stack, presenting one modally, or dismissing a screen presented in either way) would cause the the VoiceOver cursor to focus on the element in the top-left of the screen. Ideally, you would want to focus on the screen's title, or an appropriate heading/summary which would give enough context about what the user has transitioned to.

## Problem
Unfortunately, the behaviour I'm seeing is that the cursor is (for the most part) placed at the element in the same approximate location that as where it was on the previous screen. This is unexpected behaviour and leads to a sub-optimal experience for Screen Reader users.

## Solutions
Some possible solutions that I can see:
1. Listen for screen appearance inside [Header](https://github.com/react-navigation/stack/blob/master/src/views/Header/Header.tsx) and use [AccessibiltyInfo](https://facebook.github.io/react-native/docs/accessibilityinfo) to set the focus on the desired view.
2. Always pass a `headerTitle` component instead of a string when setting a screen's `navigationOptions`. Then, on the app, [listen for screen appearance](https://reactnavigation.org/docs/en/function-after-focusing-screen.html) and set the focus to the component you passed as `headerTitle`.

As a consumer, obviously I would prefer solution 1, I also think it would be great to provide this behaviour to all users of this library (perhaps behind a switch, at first?).

It's also possible that [using the native navigation components](https://reactnavigation.org/blog/2019/10/17/react-navigation-native.html) will resolve this, but I haven't tried.

## Notes
Package Versions:
```
"react-native": "0.61.4",
"react-navigation": "4.0.10",
"react-navigation-drawer": "2.3.3",
"react-navigation-stack": "1.10.3",
"react-navigation-tabs": "2.5.6",
```

**Comment by niksawtschuk** [comment]
Still an issue. Haven’t tried on v5 though.

**Comment by lissau** [comment]
Hi @niksawtschuk 

I'm developing an app where half of our users are visually impaired, so needless to say this was a big issue for us as well. 

What we do to get this behaviour is to store a reference of the header component together with the routename when the header is being rendered. We then listen for 'didFocus' events (in a hook) and change the accessibility focus to the header when a screen renders (like your option 1). It works well.

If you would like to try it out, I can give you a link to our beta.

**Comment by niksawtschuk** [comment]
@lissau that's awesome! It sounds like you're taking the second approach I mentioned, then?

I would love a link to your beta app to try it out.

**Comment by satya164** [comment]
Feel free to open issues with suggestions or send PRs for accessibility here https://github.com/react-navigation/navigation-ex
